gensim/similarities/termsim.py

```diff
@@ -315,22 +401,37 @@ def inner_product(self, X, Y, normalized=False):
 
             result = X.T.dot(matrix).dot(Y)
 
-            if normalized:
+            norm = 1.0
+
+            if normalized_X:
```

Comment:
If we're doing the same thing twice here, then it's better to make it a function and re-use it.

Usually, the rule is "three or more" but in this case the code is sufficiently complicated (at least to me as the maintainer) to warrant simplifying it.

Reply:
Fixed in c42b7cc8.